baseband/vlbi_base/base.py

```diff
@@ -109,45 +89,70 @@ class initializer.
         """
         return self._subset
 
-    def _get_sample_shape(self, subset_wrapints):
-        # Extract sample_shape by creating a dummy sample and indexing it
-        # with subset_wrapints.
-        dummy_sample = np.empty(self._unsliced_shape)
+    def _get_sample_shape(self):
+        # Extract semple_shape by creating a possibly squeezed dummy sample,
+        # and then indexing it with subset.
+        sample_shape = self._unsliced_shape
+        fields = getattr(sample_shape, '_fields', None)
+        if self.squeeze:
+            squeezed_shape = tuple(dim for dim in sample_shape if dim > 1)
+            if fields is None:
+                sample_shape = squeezed_shape
+            else:
+                fields = [field for field, dim in zip(fields, sample_shape)
+                          if dim > 1]
+                shape_cls = namedtuple('SampleShape', ','.join(fields))
+                sample_shape = shape_cls(*squeezed_shape)
+
+        if not self.subset:
+            return sample_shape
+
+        # Create a dummy sample that has the sample number as its value.
+        dummy_data = np.arange(13.)
+        dummy_sample = np.rollaxis(
+            (np.zeros(sample_shape)[..., np.newaxis] + dummy_data), -1)
         try:
-            dummy_subsample = dummy_sample[subset_wrapints]
-        except IndexError as exc:
-            exc.args += ("subset cannot be used to set sample shape.",)
+            dummy_subset = dummy_sample[(slice(None),) + self.subset]
+            # Sanity checks on result.  Sample numbers should be preserved.
+            assert np.all(dummy_subset == dummy_data.reshape(
```

Comment:
Are there sufficiently strange subsets that we would be worried about changing the sample order?  Right now different axes are still handled by different entries in the tuple.  (Though I suppose we might as well check.)

Reply:
Yes, I couldn't think of any either, but decided we might as well check - this is done only once here.